src/alert-prelude.c

```diff
@@ -724,25 +724,24 @@ static void PacketToDataProtoTLS(const Packet *p, const PacketAlert *pa, idmef_a
 static void PacketToDataProtoSSH(const Packet *p, const PacketAlert *pa, idmef_alert_t *alert)
 {
     json_t *js, *s_js;
-    SshState *ssh_state = (SshState *)FlowGetAppState(p->flow);
+    void *ssh_state = FlowGetAppState(p->flow);
 
     if (ssh_state == NULL)
         return;
 
-    js = json_object();
-    if (js == NULL)
+    void *tx_ptr = rs_ssh_state_get_tx(ssh_state, 0);
```

Comment:
should we test for failure here?

Reply:
Well, for SSH, if there is a not NULL state, there is one and only one transaction (see implementation of `rs_ssh_state_get_tx`).

In the C parser, there was only one structure for both state and transaction.

So, should we `BUG_ON` to have no performance cost on production and enforce this condition during tests ?

Reply:
`BUG_ON` sounds good to me.

Reply:
ok for `BUG_ON` then